### ISSUE TITLE
Bug in text-input datepicker: keyboard focus cant reach modal

### ISSUE THREAD

**Issue by Josvanegmond** [Vuepic/vue-datepicker#1272]
**Describe the bug**
When navigating with keyboard in a text-input mode calendar, the modal opens only after editting. But more importantly, the modal is only reachable by tab when you enter an invalid date. Otherwise it just closes.

**To Reproduce**
1. Go to a datepicker with text-input using tab
2. Edit the date so that it becomes invalid
3. Press tab, the focus is now at an unknown location (another bug)
4. Press tab again, now it is in the modal

Repeat but now enter a valid date. Press tab, and the modal closes, you can't reach it.

Additionally, if arrow-navigation is enabled, the modal steals the focus from the input and you have the reverse problem: you cant edit the text.

**Expected behavior**
Modal opening should be consistent between mouse and keyboard focus.
Focus should always clearly be visible where it is.
The modal should always be reachable by keyboard.

**Screenshots**
Easily testable at https://vue3datepicker.com/props/modes/#text-input

**Desktop & mobile (please complete the following information):**
 - Browser: Chrome
 - Library version: 12.1.0

**Comment by Jasenkoo** [comment]
This should be fixed in v13
